### DISCUSSION TITLE
drumgizmo alsamidi

### DISCUSSION THREAD

**Discussion by jstma** [corrados/edrumulus#71] [General]
@corrados I noticed that you added aslamidi input engine to drumgizmo:

https://linuxmusicians.com/viewtopic.php?t=22714

I downloaded the latest from drumgizmo and compiled it.  There is no alsamidi option.  Do you know the status?  It seems it was merged.

**Comment by corrados** [comment]
I haven't used the ALSA MIDI interface for a long while. I am using the Jack interface instead.
The ALSA MIDI should be [configured as YES per default](http://git.drumgizmo.org/drumgizmo.git/tree/configure.ac?h=develop#n422). So, it should be available. How did you start Drumgizmo, i.e., what are your command line parameters?

**Reply by jstma** [reply]
I have a script that "works":

`$PROG -i jackmidi -I midimap=$KIT/midimap.xml -o alsa $KIT/ShittyKit.xml`

But I can't see the midi interface anywhere.  If I try this:

`$PROG -i alsamidi -I midimap=$KIT/midimap.xml -o alsa $KIT/ShittyKit.xml`

It shows me the help.  After I ran ./configure there was no mention of alsamidi being yes or no.

I grabbed the source from here:

http://www.drumgizmo.org/releases/drumgizmo-0.9.20/drumgizmo-0.9.20.tar.gz

which seems to be the latest release.

**Reply by jstma** [reply]
Ok I don't know what was going on last night when I posted the question, but as of this morning:

`$PROG -i alsamidi -I midimap=$KIT/midimap.xml -o alsa $KIT/ShittyKit.xml`

works just fine.  At least it creates an alsa midi port, I haven't tested beyond that.

I must have mistakenly been using the drumgizmo I installed through apt-get which is a very old version.

**Reply by corrados** [reply]
Good to hear that it works now :+1:
